tests/test_asyncio.py

```diff
@@ -51,3 +49,24 @@ async def test_async_docarray_preserve_original_inputs(
     assert '__loaded_by_CAS__' not in t2.result()[0].tags
     assert not t2.result()[0].tensor
     assert not t2.result()[0].blob
+    assert inputs[0] is t2.result()[0]
+
+
+@pytest.mark.parametrize(
+    'inputs',
+    [
+        [Document(text='hello, world') for _ in range(20)],
+        DocumentArray([Document(text='hello, world') for _ in range(20)]),
+    ],
+)
+@pytest.mark.asyncio
+async def test_async_docarray_preserve_original_order(make_flow, inputs):
+    c = Client(server=f'grpc://0.0.0.0:{make_flow.port}')
+    t1 = asyncio.create_task(another_heavylifting_job())
+    t2 = asyncio.create_task(
+        c.aencode(inputs if not callable(inputs) else inputs(), batch_size=1)
+    )
+    await asyncio.gather(t1, t2)
+    assert isinstance(t2.result(), DocumentArray)
+    for i in range(len(inputs)):
+        assert inputs[i] is t2.result()[i]
```

Comment:
I recommend using some `id` to denote the order, which will make this test more clear

Reply:
I use `is` in a loop to make sure the modification is in-place

Reply:
Then there should be another `test_in_place` something like this

Reply:
done